qcodes/instrument_drivers/tektronix/TPS2012.py

```diff
@@ -0,0 +1,243 @@
+
+            self.add_parameter('ch{}_curvedata'.format(ch),
+                               label='Curve data',
+                               units='V',
+                               get_cmd=partial(curveasker, ch),
+                               get_parser=self._curveparameterparser)
+
+        # Necessary settings for parsing the binary curve data
+        self.visa_handle.encoding = 'latin-1'
+        log.info('Set VISA encoding to latin-1')
+        self.write('DATa:ENCdg RPBinary')
+        log.info('Set TPS2012 data encoding to RPBinary' +
+                 ' (Positive Integer Binary)')
+        self.write('DATa:WIDTh 2')
+        log.info('Set TPS2012 data width to 2')
+        # Note: using data width 2 has been tested to not add
+        # significantly to transfer times. The maximal length
+        # of an array in one transfer is 2500 points.
+
+    ##################################################
+    # HELPFUL PARSER FUNCTIONS                       #
+    ##################################################
+
+    def _binaryparser(self, curve):
+        """
+        Helper function for parsing the curve data
+
+        Args:
+            curve (str): the return value of 'CURVe?' when
+            DATa:ENCdg is set to RPBinary.
+            Note: The header and findal newline character
+            must be removed.
+
+        Returns:
+            nparray: the curve in units where the digitisation range
+            is mapped to (-32768, 32767).
+        """
+        # TODO: Add support for data width = 1 mode?
+        output = np.zeros(int(len(curve)/2))  # data width 2
+        # output = np.zeros(int(len(curve)))  # data width 1
+        for ii in range(len(output)):
+            # casting FTW
+            temp = curve[2*ii:2*ii+1].encode('latin-1')  # data width 2
+            temp = binascii.b2a_hex(temp)
+            temp = (int(temp, 16)-128)*256  # data width 2 (1)
+            output[ii] = temp
+        return output
+
+    def _preambleparser(self, response):
+        """
+        Parser function for the curve preamble
+
+        Args:
+            preamble (str): The reponse of WFMPre?
+
+        Returns:
+            dict: a dictionary containing the following keys:
+              no_of_bytes, no_of_bits, encoding, binary_format,
+              byte_order, no_of_points, waveform_ID, point_format,
+              x_incr, x_zero, x_unit, y_multiplier, y_zero, y_offset, y_unit
+        """
+        response = response.split(';')
+        outdict = {}
+        outdict['no_of_bytes'] = response[0]
+        outdict['no_of_bits'] = response[1]
+        outdict['encoding'] = response[2]
+        outdict['binary_format'] = response[3]
+        outdict['byte_order'] = response[4]
+        outdict['no_of_points'] = response[5]
+        outdict['waveform_ID'] = response[6]
+        outdict['point_format'] = response[7]
+        outdict['x_incr'] = response[8]
+        # outdict['point_offset'] = response[9]  # Always zero
```

Comment:
I intentionally left it there as an explanation to curious souls; what about field 9? If you prefer, I can reformulate the line.